include/libkrun.h

```diff
@@ -113,6 +120,41 @@ int32_t krun_set_data_disk(uint32_t ctx_id, const char *disk_path);
  */
 int32_t krun_add_disk(uint32_t ctx_id, const char *block_id, const char *disk_path, bool read_only);
 
+/* Supported disk image formats */
+#define KRUN_DISK_FORMAT_UNSPECIFIED 0
+#define KRUN_DISK_FORMAT_RAW 1
+#define KRUN_DISK_FORMAT_QCOW2 2
+/**
+ * Adds a disk image to be used as a general partition for the microVM. The supported
+ * image formats are: "raw" and "qcow2". 
+ *
+ * This API is mutually exclusive with the deprecated krun_set_root_disk and
+ * krun_set_data_disk methods and must not be used together.
+ *
+ * By all possible means, one should avoid probing the disk image that will be given to libkrun.
+ * If a raw file is probed, and a QCOW2 header is written into the raw file's header, the next time
+ * libkrun launches the VM it will boot as a QCOW2 image. QCOW2 files can, and will, open other files.
+ * The guest will most likely have access to the data in those files, essentially giving the guest
+ * access to any file on the host system the disk emulation has access to.
+ *
+ * Arguments:
+ *  "ctx_id"      - the configuration context ID.
+ *  "block_id"    - a null-terminated string representing the partition.
+ *  "disk_path"   - a null-terminated string representing the path leading to the disk image that
+ *                  contains the root file-system.
```

Comment:
This doesn't have to be the "root file-system" though, does it? Since this function replaces `krun_set_data_disk`...

Reply:
Fixed.